hooks/plugin_base.py

```diff
@@ -0,0 +1,133 @@
+        return description_group_box
+
+    def get_ui_settings(self, widget):
+        """
+        Invoked by the publisher when the selection changes so the new settings
+        can be applied on the previously selected tasks.
+
+        The widget argument is the widget that was previously created by
+        `create_settings_widget`.
+
+        The method returns an dictionary, where the key is the name of a
+        setting that should be updated and the value is the new value of that
+        setting. Note that it is not necessary to return all the values from
+        the UI. This is to allow the publisher to update a subset of settings
+        when multiple tasks have been selected.
+
+        Example::
+
+            {
+                 "setting_a": "/path/to/a/file"
+            }
+
+        :param widget: The widget that was created by `create_settings_widget`
+        """
+
+        # the default implementation does not show any editable widgets, so this
+        # is a no-op. this method is required to be defined in order for the
+        # custom UI to show up in the app
+        return {}
+
+    def set_ui_settings(self, widget, settings):
+        """
+        Allows the custom UI to populate its fields with the settings from the
+        currently selected tasks.
+
+        The widget is the widget created and returned by
+        `create_settings_widget`.
+
+        A list of settings dictionaries are supplied representing the current
+        values of the settings for selected tasks. The settings dictionaries
+        correspond to the dictionaries returned by the settings property of the
+        hook.
+
+        Example::
+
+            settings = [
+            {
+                 "seeting_a": "/path/to/a/file"
+                 "setting_b": False
+            },
+            {
+                 "setting_a": "/path/to/a/file"
+                 "setting_b": False
+            }]
+
+        The default values for the settings will be the ones specified in the
+        environment file. Each task has its own copy of the settings.
+
+        When invoked with multiple settings dictionaries, it is the
+        responsibility of the custom UI to decide how to display the
+        information. If you do not wish to implement the editing of multiple
+        tasks at the same time, you can raise a `NotImplementedError` when
+        there is more than one item in the list and the publisher will inform
+        the user than only one task of that type can be edited at a time.
+
+        :param widget: The widget that was created by `create_settings_widget`
+        :param settings: a list of dictionaries of settings for each selected
+            task.
+        """
+
+        # the default implementation does not show any editable widgets, so this
+        # is a no-op. this method is required to be defined in order for the
+        # custom UI to show up in the app
+        pass
```

Comment:
to complete the interface, should we add accept-validate etc methods too? With comments?

Reply:
Yeah, I was thinking about that. This seems like the file to use when generating the docs. I will update.